src/contracts/GPv2AccessControl.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: LGPL-3.0-or-newer
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+
+/// @title Gnosis Protocol v2 Access Control Contract
+/// @author Gnosis Developers
+contract GPv2AccessControl is Ownable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    EnumerableSet.AddressSet private solvers;
+
+    function addSolver(address solverAddress) public {
+        solvers.add(solverAddress);
+    }
+
+    function isSolver(address prospectiveSolver) public view returns (bool) {
+        return solvers.contains(prospectiveSolver);
+    }
+
+    function getSolverAt(uint256 index) public view returns (address) {
+        return solvers.at(index);
+    }
+
+    function numSolvers() public view returns (uint256) {
+        return solvers.length();
+    }
```

Comment:
Looking at these methods it could be a good use case for using the `StorageAccessible` base class so we can add view methods later on.

Reply:
Cool idea. If we plan to do this, then I might expect the view methods already included here should/could be removed for now.

Reply:
Hey @fleupold - just want to know if you think these methods should be removed for now in favour of using storage accessible?

Reply:
Since this will require more work than originally expected, I have created a separate issue (#172) for this.